src/app/complex/complex-results/complex-results.component.ts

```diff
@@ -5,39 +5,56 @@ import {ComplexPortalService} from '../shared/service/complex-portal.service';
 import {ProgressBarComponent} from '../../shared/loading-indicators/progress-bar/progress-bar.component';
 import {Title} from '@angular/platform-browser';
 import {AnalyticsService} from '../../shared/google-analytics/service/analytics.service';
-
+import {Interactor} from '../shared/model/complex-results/interactor.model';
+import {NotificationService} from '../../shared/notification/service/notification.service';
 
 @Component({
   selector: 'cp-complex-results',
   templateUrl: './complex-results.component.html',
-  styleUrls: ['./complex-results.component.css']
+  styleUrls: ['./complex-results.component.css'],
 })
 export class ComplexResultsComponent implements OnInit, AfterViewInit {
+  LIST_VIEW = 'view_list';
+  COMPLEX_NAVIGATOR_VIEW = 'view_complex_navigator';
   private _query: string;
-  private _currentPageIndex: number;
   private _complexSearch: ComplexSearchResult;
-  private _lastPageIndex: number;
-  private _pageSize = 10;
   private _spicesFilter: string[];
   private _bioRoleFilter: string[];
   private _interactorTypeFilter: string[];
+  private _allInteractorsInComplexSearch: Interactor[] = [];
+  DisplayType: string;
+  toast;
+
+  private _listPageSize = 15; // This is where we set the size of the pages for list view
+  private _navigatorPageSize = 20; // This is where we set the size of the pages for navigator view
+  private _listCurrentPage: number;
+  private _navigatorCurrentPage: number;
+  private _listLastPageIndex;
+  private _navigatorLastPageIndex;
 
   constructor(private route: ActivatedRoute, private router: Router,
               private complexPortalService: ComplexPortalService, private titleService: Title,
-              private googleAnalyticsService: AnalyticsService) {
+              private googleAnalyticsService: AnalyticsService, private notificationService: NotificationService) {
   }
 
   ngOnInit() {
     this.titleService.setTitle('Complex Portal - Results');
-
+    this._allInteractorsInComplexSearch = [];
+    this.route.fragment.subscribe(fragment => {
+      if (fragment === this.COMPLEX_NAVIGATOR_VIEW) {
+        this.DisplayType = this.COMPLEX_NAVIGATOR_VIEW;
+      } else if (fragment === this.LIST_VIEW) {
+        this.DisplayType = this.LIST_VIEW;
+      }
+    });
     this.route
```

Comment:
Here I didn't mean to move `this.route.queryParams.subscribe` after `this.route.fragment.subscribe`. I meant moving it inside. The potential issue with the 2 subscribers separate is that they are synchronous, so we cannot make sure of the order they run, so we may be checking the query params before the fragment and we shouldn't as there are variables, like page index, that depend on the fragment. So, to avoid that we can do

```
    this.route.fragment.subscribe(fragment => {
      if (fragment === this.COMPLEX_NAVIGATOR_VIEW) {
        this.DisplayType = this.COMPLEX_NAVIGATOR_VIEW;
      } else if (fragment === this.LIST_VIEW) {
        this.DisplayType = this.LIST_VIEW;
      }
      this.route
        .queryParams
        .subscribe(queryParams => {
            ...
        });
    });
```

So, after processing the fragment, we process the query params.